### PR TITLE
NLogTraceListener - Reduce allocations for TraceData with single param

### PR DESCRIPTION
**Pull request by snakefoot**
And handle possible null-reference-exception

### CODE REVIEW

File: src/NLog/NLogTraceListener.cs [hunk 1]
```diff
@@ -251,20 +248,32 @@ public override void TraceData(TraceEventCache eventCache, string source, TraceE
...
             {
-                if (i > 0)
+                if (data.Length == 1)
                 {
-                    sb.Append(", ");
+                    message = "{0}";
+                }
+                else
+                {
+                    var sb = new StringBuilder();
```
Comment (304NotModified) [inline src/NLog/NLogTraceListener.cs]: maybe set also the initial capacity? (5*(i+1))

File: src/NLog/NLogTraceListener.cs [hunk 2]
```diff
@@ -251,20 +248,32 @@ public override void TraceData(TraceEventCache eventCache, string source, TraceE
...
+                        if (i > 0)
+                        {
+                            sb.Append(", ");
+                        }
+
+                        sb.Append("{");
+                        sb.Append(i);
+                        sb.Append("}");
+                    }
+                    message = sb.ToString();
```
Comment (304NotModified) [inline src/NLog/NLogTraceListener.cs]: also, isn't string.join more efficient in this way?

File: src/NLog/NLogTraceListener.cs (hunk 1 above)
Comment (snakefoot) [reply src/NLog/NLogTraceListener.cs]: Yes good idea.

File: src/NLog/NLogTraceListener.cs (hunk 2 above)
Comment (snakefoot) [reply src/NLog/NLogTraceListener.cs]: Not sure I understand string.Join, but I have now changed to use StringBuilderExt.AppendInvariant

Comment (304NotModified) [reply src/NLog/NLogTraceListener.cs]: I was wrong, but for the reference: https://stackoverflow.com/questions/585860/string-join-vs-stringbuilder-which-is-faster#585897

**Comment by 304NotModified** [comment]
Thanks!

**Pull request merged**